### PR TITLE
[Merged by Bors] - chore: Refactor Chinese remainders

### PR DESCRIPTION
**Pull request by PatrickMassot**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Basic.lean [hunk 1]
```diff
@@ -805,6 +805,13 @@ theorem prod_eq_single {s : Finset α} {f : α → β} (a : α) (h₀ : ∀ b 
 #align finset.prod_eq_single Finset.prod_eq_single
 #align finset.sum_eq_single Finset.sum_eq_single
 
+@[to_additive]
+lemma prod_univ_eq_single [Fintype α] {f : α → β} (a : α)
```
Comment (eric-wieser) [inline Mathlib/Algebra/BigOperators/Basic.lean:809]: I think we might already have this in the `Fintype` namespace.

File: Mathlib/RingTheory/Ideal/Operations.lean [hunk 2]
```diff
@@ -875,6 +875,24 @@ theorem isCoprime_span_singleton_iff (x y : R) :
   · rintro ⟨a, _, ⟨b, rfl⟩, e⟩; exact ⟨a, b, mul_comm b y ▸ e⟩
   · rintro ⟨a, b, e⟩; exact ⟨a, _, ⟨b, rfl⟩, mul_comm y b ▸ e⟩
 
+theorem coprime_iInf_of_coprime {J : ι → Ideal R} {s : Finset ι}
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Ideal/Operations.lean:878]:
```suggestion
theorem isCoprime_biInf {J : ι → Ideal R} {s : Finset ι}
```
Though it's probably a good idea to have the `Fintype` version too.

File: Mathlib/Algebra/BigOperators/Basic.lean (hunk 1 above)
Comment (PatrickMassot) [reply Mathlib/Algebra/BigOperators/Basic.lean:809]: I was very surprised but couldn't find it. But I just searched again after `import Mathlib` and indeed it exists.

File: Mathlib/RingTheory/Ideal/Operations.lean [hunk 3]
```diff
@@ -875,6 +875,24 @@ theorem isCoprime_span_singleton_iff (x y : R) :
   · rintro ⟨a, _, ⟨b, rfl⟩, e⟩; exact ⟨a, b, mul_comm b y ▸ e⟩
   · rintro ⟨a, b, e⟩; exact ⟨a, _, ⟨b, rfl⟩, mul_comm y b ▸ e⟩
 
+theorem isCoprime_biInf {J : ι → Ideal R} {s : Finset ι}
+    (hf : ∀ j ∈ s, IsCoprime I (J j)) : IsCoprime I (⨅ j ∈ s, J j) := by
+  classical
+  simp_rw [isCoprime_iff_add] at *
+  revert hf
+  induction s using Finset.induction with
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Ideal/Operations.lean:883]: I think `induction` does this `revert` automatically?

Comment (PatrickMassot) [reply Mathlib/RingTheory/Ideal/Operations.lean]: Good catch!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Comment by ADedecker** [comment]
bors retry

**Pull request closed without merging**